src/Raven.Server/Documents/Handlers/Debugging/ServerWideDebugInfoPackageHandler.cs

```diff
@@ -309,11 +311,24 @@ private async Task WriteServerInfo(ZipArchive archive, JsonOperationContext json
 
             foreach (var route in routes)
             {
-                token.ThrowIfCancellationRequested();
-
+                Exception ex = null;
                 var sw = Stopwatch.StartNew();
-                await InvokeAndWriteToArchive(archive, context, localEndpointClient, route, path, endpointParameters, token);
-                debugInfoDict[route.Path] = sw.Elapsed;
+                try
+                {
+                    token.ThrowIfCancellationRequested();
+
+                    await InvokeAndWriteToArchive(archive, context, localEndpointClient, route, path, endpointParameters, token);
+                    debugInfoDict[route.Path] = sw.Elapsed;
+                }
+                catch (Exception e)
+                {
+                    ex = e;
+                }
+                finally
+                {
+                    if (_logger.IsOperationsEnabled)
+                        _logger.Operations($"Elapsed time for '{route.Path}': {sw.Elapsed} ", ex);
```

Comment:
Why is that as an `operation` level? And why as separate messages?

Reply:
This does not help us track hanging requests.
There should be a message before executing and after.

Do we want to introduce a timeout per endpoint? Even big one like 5 or 10min? cc @ayende

Reply:
Yes, a timeout of 1 minute should be enough, I think.
Do we have anything that _should_ take more than that?

Reply:
done